crates/iota-types/src/iota_system_state/iota_system_state_summary.rs

```diff
@@ -20,14 +20,186 @@ use crate::{
+/// This is the JSON-RPC type for IOTA system state objects.
+/// It is an enum type that can represent either V1 or V2 system state objects.
+pub enum IotaSystemStateSummary {
+    V1(IotaSystemStateSummaryV1),
+    V2(IotaSystemStateSummaryV2),
+}
+
+/// This is the JSON-RPC type for the
+/// [`IotaSystemStateV1`](super::iota_system_state_inner_v1::IotaSystemStateV1)
+/// object. It flattens all fields to make them top-level fields such that it as
+/// minimum dependencies to the internal data structures of the IOTA system
+/// state type.
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct IotaSystemStateSummaryV1 {
+    /// The current epoch ID, starting from 0.
+    #[schemars(with = "BigInt<u64>")]
+    #[serde_as(as = "Readable<BigInt<u64>, _>")]
+    pub epoch: u64,
+    /// The current protocol version, starting from 1.
+    #[schemars(with = "BigInt<u64>")]
+    #[serde_as(as = "Readable<BigInt<u64>, _>")]
+    pub protocol_version: u64,
+    /// The current version of the system state data structure type.
+    #[schemars(with = "BigInt<u64>")]
+    #[serde_as(as = "Readable<BigInt<u64>, _>")]
+    pub system_state_version: u64,
+    /// The current IOTA supply.
+    #[schemars(with = "BigInt<u64>")]
+    #[serde_as(as = "Readable<BigInt<u64>, _>")]
+    pub iota_total_supply: u64,
+    /// The `TreasuryCap<IOTA>` object ID.
+    pub iota_treasury_cap_id: ObjectID,
+    /// The storage rebates of all the objects on-chain stored in the storage
+    /// fund.
+    #[schemars(with = "BigInt<u64>")]
+    #[serde_as(as = "Readable<BigInt<u64>, _>")]
+    pub storage_fund_total_object_storage_rebates: u64,
+    /// The non-refundable portion of the storage fund coming from
+    /// non-refundable storage rebates and any leftover
+    /// staking rewards.
+    #[schemars(with = "BigInt<u64>")]
+    #[serde_as(as = "Readable<BigInt<u64>, _>")]
+    pub storage_fund_non_refundable_balance: u64,
+    /// The reference gas price for the current epoch.
+    #[schemars(with = "BigInt<u64>")]
+    #[serde_as(as = "Readable<BigInt<u64>, _>")]
+    pub reference_gas_price: u64,
+    /// Whether the system is running in a downgraded safe mode due to a
+    /// non-recoverable bug. This is set whenever we failed to execute
+    /// advance_epoch, and ended up executing advance_epoch_safe_mode.
+    /// It can be reset once we are able to successfully execute advance_epoch.
+    pub safe_mode: bool,
+    /// Amount of storage charges accumulated (and not yet distributed) during
+    /// safe mode.
+    #[schemars(with = "BigInt<u64>")]
+    #[serde_as(as = "Readable<BigInt<u64>, _>")]
+    pub safe_mode_storage_charges: u64,
+    /// Amount of computation charges accumulated (and not yet distributed)
+    /// during safe mode.
+    #[schemars(with = "BigInt<u64>")]
+    #[serde_as(as = "Readable<BigInt<u64>, _>")]
+    pub safe_mode_computation_charges: u64,
```

Comment:
Can we revert to `safe_mode_computation_rewards` for V1? This way `IotaSystemStateSummaryV1` would be exactly the same as the former `IotaSystemStateSummary` which is used in the public interfaces.

Reply:
Sure, done.